### PR TITLE
To fix webconfig coverity defects

### PR DESCRIPTION
**Pull request by sadhyama**

### CODE REVIEW

File: source/WebConfig/webconfig_internal.c [hunk 1]
```diff
@@ -573,8 +574,9 @@ int requestWebConfigData(char **configData, int r_count, int index, int status,
 		}
 
 		// set callback for writing received data
+		dataVal = &data;
 		curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_callback_fn);
-		curl_easy_setopt(curl, CURLOPT_WRITEDATA, &data);
+		curl_easy_setopt(curl, CURLOPT_WRITEDATA, dataVal);
```
Comment (shilpa24balaji) [inline source/WebConfig/webconfig_internal.c]: void * typecast also should work, variable is not needed

File: source/WebConfig/webconfig_internal.c [hunk 2]
```diff
@@ -1404,6 +1419,7 @@ void* processWebConfigNotification()
 		msg = notifyMsg;
 		if(msg !=NULL)
 		{
+			pthread_mutex_unlock (&notify_mut);
```
Comment (shilpa24balaji) [inline source/WebConfig/webconfig_internal.c]: msg and notifyMsg are modified below, so unlock cannot be done here.

File: source/WebConfig/cosa_webconfig_dml.c [hunk 3]
```diff
@@ -255,6 +255,7 @@ ConfigFile_Synchronize
     ULONG                                 entryCount        = 0;
 	WebcfgDebug("------- %s ----- ENTER ----\n",__FUNCTION__);
 	WebcfgDebug("------- %s ----- EXIT ----\n",__FUNCTION__);
+    return 0;
```
[the remark this answers is not in the document]
Comment (sadhyama) [reply source/WebConfig/cosa_webconfig_dml.c]: Incorporated.

File: source/WebConfig/cosa_webconfig_apis.h [hunk 4]
```diff
@@ -139,3 +139,9 @@ CosaDmlRemoveConfigFileEntry
 
 
 int initConfigFileWithURL(char *Url, ULONG InstanceNumber);
+
+void RemoveEntryFromIndexesList(ULONG InstanceNumber);
```
[the remark this answers is not in the document]
Comment (sadhyama) [reply source/WebConfig/cosa_webconfig_apis.h]: Incorporated.

File: source/WebConfig/webconfig_internal.c (hunk 2 above)
Comment (sadhyama) [reply source/WebConfig/webconfig_internal.c]: Incorporated.

File: source/WebConfig/webconfig_internal.c (hunk 1 above)
Comment (sadhyama) [reply source/WebConfig/webconfig_internal.c]: (void*)&data doesn't help, coverity still shows **BAD_SIZEOF** issue.
_Event bad_sizeof: Taking the size of "&data", which is the address of an object, is suspicious._
Hence retained same above approach.

File: source/WebConfig/webconfig_internal.c [hunk 5]
```diff
@@ -1453,11 +1468,10 @@ void* processWebConfigNotification()
 				if(msg != NULL)
 				{
 					free_notify_params_struct(msg);
+					msg = NULL;
 					notifyMsg = NULL;
 				}
 			}
-			pthread_mutex_unlock (&notify_mut);
```
Comment (shilpa24balaji) [inline source/WebConfig/webconfig_internal.c]: This should not be deleted

File: source/WebConfig/cosa_webconfig_apis.h [hunk 6]
```diff
@@ -139,3 +139,8 @@ CosaDmlRemoveConfigFileEntry
 
 
 int initConfigFileWithURL(char *Url, ULONG InstanceNumber);
+
+void CosaDmlRemoveValueFromDb(char *ParamName);
+void CosaDmlStoreValueIntoDb(char *ParamName, char *pString);
+void CosaDmlGetValueFromDb( char* ParamName, char* pString );
+void FillEntryInList(PCOSA_DATAMODEL_WEBCONFIG pWebConfig, PCOSA_DML_WEBCONFIG_CONFIGFILE_ENTRY configFileEntry);
```
Comment (shilpa24balaji) [inline source/WebConfig/cosa_webconfig_apis.h]: Check all above functions and only if needed expose in header

File: source/WebConfig/webconfig_internal.c (hunk 5 above)
Comment (sadhyama) [reply source/WebConfig/webconfig_internal.c]: done.

File: source/WebConfig/cosa_webconfig_apis.h (hunk 6 above)
Comment (sadhyama) [reply source/WebConfig/cosa_webconfig_apis.h]: Incorporated.

**Pull request merged**